### PR TITLE
Server annotations

### PR DESCRIPTION
**Pull request by bryevdv**
This PR cleans up property warnings, etc around tool annotations mostly. 

Seems to silence issues with box and poly selection overlays as well as tickers on datetime tickers.

### CODE REVIEW

**Comment by bryevdv** [comment]
I also removed the histogram charts from the stocks example for now because they seem to be breaking it badly.

File: bokeh/models/annotations.py [hunk 1]
```diff
@@ -220,3 +222,21 @@ class Span(Annotation):
...
     The %s values for the span.
     """)
+
+class Tooltip(Annotation):
+    """ Render a tooltip.
+
+    .. note::
+        This model is typically managed by BokehJS and is not useful
+        directly from python.
+
```
Comment (havocp) [inline bokeh/models/annotations.py]: it isn't clear to me what this note means - does it mean setting the properties on this won't work?
or just that I probably don't want to set them?

Comment (bryevdv) [inline bokeh/models/annotations.py]: I mean we don't actually (currently) ever create or set a tooltip from python, but the class needs to exist to please our machinery. I'm not sure we will either, on the JS side it creates a tooltip per renderer. I think we need a concept of "client originated" models .

File: bokehjs/src/coffee/renderer/annotation/span.coffee [hunk 2]
```diff
@@ -14,13 +13,21 @@ class SpanView extends PlotWidget
     @$el.hide()
 
   bind_bokeh_events: () ->
-    @listenTo(@model, 'change:location', @_draw_span)
+    if @mget('for_hover')
```
Comment (havocp) [inline bokehjs/src/coffee/renderer/annotation/span.coffee]: this seems like it won't update if for_hover changes, fwiw

Comment (bryevdv) [inline bokehjs/src/coffee/renderer/annotation/span.coffee]: My intent was really that this was a initialization configuration, maybe there is a better way to express that than making it a BB property.

**Comment by bryevdv** [comment]
Also, I regard this as a bit of a bandaid --- I'd like to rework things so that tool overlays just use and are configured with `BoxAnnotation` and `PolyAnnotation` which would simplify things and also open up more configuration for users. But that would be easier for me starting from this PR merged as a basis.

**Pull request merged**